include/pagmo/algorithms/nsga2.hpp

```diff
@@ -534,10 +535,12 @@ class nsga2
 
         // This implements the integer mutation for an individual
         for (decltype(D) j = Dc; j < D; ++j) {
-            if (drng(m_e) <= m_m) {
-                std::uniform_int_distribution<vector_double::size_type> ra_num(
-                    static_cast<vector_double::size_type>(lb[j]), static_cast<vector_double::size_type>(ub[j]));
-                child[j] = static_cast<double>(ra_num(m_e));
+            if (drng(m_e) < m_m) {
+                // We need to draw a random integer in [lb, ub]. Since these are floats we
+                // cannot use integer distributions without risking overflows, hence we use the 
+                // pagmo utility which takes care of it random_decision_vector
+                auto mutated = random_decision_vector({{lb[j]},{ub[j]}}, m_e, 1u);
```

Comment:
This will do a memory allocation at every invocation of ``random_decision_vector()``. Are we ok with the performance implications of this? Is this code gonna be executed very frequently?

A solutio would be to have an additiona overload of ``random_decision_vector()`` that takes ``mutated`` (the return value) as a first parameter by preference.

Reply:
thought about it, indeed it gets called a lot, but not sure would be a bottle neck ....

Reply:
I can just code it on the fly without invoking the utility

Reply:
I can just put the code
```
        if (!std::isfinite(ub + 1.)) {
            pagmo_throw(std::invalid_argument,
                        "The upper bound of the integer part is too large for the decision vector index "
                            + std::to_string(i));
        }

        mutated = uniform_real_from_range(lb, ub + 1, m_e);
```

Reply:
You need the floor as well right? And isn't ``uniform_real_from_range()`` already checking for infinity in the bounds?

Reply:
the floor is there to get an int out of a float and ensure the drawn distribution is uniform from lb to ub. We use the same trick in ``random_decision_vector``

The infinity bounds are indeed checked in ``uniform_real_from_range`` so maybe we should also remove that check in ``random_decision_vector``?

Reply:
Yep it looks like the check should be removed in ``random_decision_vector()`` as well.